commands/scanandfixrepos.go

```diff
@@ -33,12 +36,26 @@ func (saf *ScanAndFixRepositories) Run(configAggregator utils.FrogbotConfigAggre
 
 func (saf *ScanAndFixRepositories) scanAndFixSingleRepository(repoConfig *utils.FrogbotRepoConfig, client vcsclient.VcsClient) error {
 	for _, branch := range repoConfig.Branches {
-		err := saf.downloadAndRunScanAndFix(repoConfig, branch, client)
+		shouldScan, checkedCommit, err := saf.shouldScanLatestCommit(context.Background(), repoConfig, client, branch)
+		if err != nil {
+			return err
+		}
+		if !shouldScan {
+			log.Info(fmt.Sprintf("Commit %s has already been scanned by FrogBot", checkedCommit))
+			continue
+		}
+		err = saf.downloadAndRunScanAndFix(repoConfig, branch, client)
+		if err != nil {
```

Comment:
The above 2 lines can be rediced to the following single line - 
```
if err = saf.downloadAndRunScanAndFix(repoConfig, branch, client); err != nil {
```
The above applies to the other if statement in this function and also for the `setCommitBuildStatus` function.